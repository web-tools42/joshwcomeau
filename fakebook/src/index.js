import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import configureStore from './store';

import App from './components/App';
import Header from './components/Header';
import Home from './components/Home';
import Profile from './components/Profile';
import WindowEvents from './components/WindowEvents';
import DevTools from './components/DevTools';


const store = configureStore();

render(
  <Provider store={store}>
    <Router>
      <App>
        <Header />

        <Route exact path="/" component={Home} />
        <Route path="/:userName" component={Profile} />

        <WindowEvents />

        <DevTools />
      </App>
    </Router>
  </Provider>,
  document.getElementById('root')
);
