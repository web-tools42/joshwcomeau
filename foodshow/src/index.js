import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { fetchPhotosRequest, loginSuccess, loginFailure } from './actions';
import { autoLogin } from './utils/unsplash.utils';

import App from './components/App';
import Slideshow from './components/Slideshow';


const store = configureStore();

// The user could be coming back from a login request,
// or they could have persisted credentials.
// Deal with auth stuff before anything else.
autoLogin({
  callback(response) {
    if (response) {
      store.dispatch(
        response.errors
        ? loginFailure({ error: response.errors[0] })
        : loginSuccess({ user: response })
      );
    }

    store.dispatch(fetchPhotosRequest());
  },
});

render((
  <Provider store={store}>
    <App>
      <Slideshow />
    </App>
  </Provider>
), document.getElementById('root'));
