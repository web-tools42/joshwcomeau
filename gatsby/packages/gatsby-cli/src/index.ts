#!/usr/bin/env node

import "@babel/polyfill"
import semver from "semver"
import util from "util"
import createCli from "./create-cli"
import report from "./reporter"
import pkg from "../package.json"
import updateNotifier from "update-notifier"

const useJsonLogger = process.argv.slice(2).some(arg => arg.includes(`json`))

if (useJsonLogger) {
  process.env.GATSBY_LOGGER = `json`
}

// Check if update is available
updateNotifier({ pkg }).notify({ isGlobal: true })

const MIN_NODE_VERSION = `>=8.0.0`

if (!semver.satisfies(process.version, MIN_NODE_VERSION)) {
  report.panic(
    report.stripIndent(`
      Gatsby requires Node.js v8 or higher (you have ${process.version}).
      Upgrade Node to the latest stable release: https://gatsby.dev/upgrading-node-js
    `)
  )
}

process.on(`unhandledRejection`, reason => {
  // This will exit the process in newer Node anyway so lets be consistent
  // across versions and crash

  // reason can be anything, it can be a message, an object, ANYTHING!
  // we convert it to an error object so we don't crash on structured error validation
  if (!(reason instanceof Error)) {
    reason = new Error(util.format(reason))
  }

  report.panic(`UNHANDLED REJECTION`, reason)
})

process.on(`uncaughtException`, error => {
  report.panic(`UNHANDLED EXCEPTION`, error)
})

createCli(process.argv)
