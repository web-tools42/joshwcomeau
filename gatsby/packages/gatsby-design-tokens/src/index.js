export { default as borders } from "./borders"
export { default as breakpoints } from "./breakpoints"
export { default as colors } from "./colors"
export { default as fonts } from "./fonts"
export { default as fontSizes } from "./font-sizes"
export { default as fontWeights } from "./font-weights"
export { default as letterSpacings } from "./letter-spacings"
export { default as lineHeights } from "./line-heights"
export { default as mediaQueries } from "./media-queries"
export { default as radii } from "./radii"
export { default as shadows } from "./shadows"
export { default as sizes } from "./sizes"
export { default as space } from "./space"
export { default as transition } from "./transition"
export { default as zIndices } from "./z-indices"
