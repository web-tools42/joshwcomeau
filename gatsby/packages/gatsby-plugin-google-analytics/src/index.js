import React from "react"
import PropTypes from "prop-types"

const createFunctionWithTimeout = (callback, opt_timeout = 1000) => {
  let called = false
  const raceCallback = () => {
    if (!called) {
      called = true
      callback()
    }
  }
  setTimeout(raceCallback, opt_timeout)
  return raceCallback
}

function OutboundLink(props) {
  return (
    <a
      {...props}
      onClick={e => {
        if (typeof props.onClick === `function`) {
          props.onClick(e)
        }
        let redirect = true
        if (
          e.button !== 0 ||
          e.altKey ||
          e.ctrlKey ||
          e.metaKey ||
          e.shiftKey ||
          e.defaultPrevented
        ) {
          redirect = false
        }
        if (props.target && props.target.toLowerCase() !== `_self`) {
          redirect = false
        }
        if (window.ga) {
          window.ga(`send`, `event`, {
            eventCategory: `Outbound Link`,
            eventAction: `click`,
            eventLabel: props.href,
            transport: redirect ? `beacon` : ``,
            hitCallback: function() {
              if (redirect) {
                document.location = props.href
              }
            },
          })
        } else {
          if (redirect) {
            document.location = props.href
          }
        }

        return false
      }}
    />
  )
}

OutboundLink.propTypes = {
  href: PropTypes.string,
  target: PropTypes.string,
  onClick: PropTypes.func,
}

/**
 * This allows the user to create custom events within their Gatsby projects.
 *
 * @param {import('gatsby-plugin-google-analytics').CustomEventArgs} args
 * @see https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#events
 */
function trackCustomEvent({
  category,
  action,
  label,
  value,
  nonInteraction = true,
  transport,
  hitCallback,
  callbackTimeout = 1000,
}) {
  if (typeof window !== `undefined` && window.ga) {
    const trackingEventOptions = {
      eventCategory: category,
      eventAction: action,
      eventLabel: label,
      eventValue: value,
      nonInteraction: nonInteraction,
      transport,
    }

    if (hitCallback && typeof hitCallback === `function`) {
      trackingEventOptions.hitCallback = createFunctionWithTimeout(
        hitCallback,
        callbackTimeout
      )
    }

    window.ga(`send`, `event`, trackingEventOptions)
  }
}

export { OutboundLink, trackCustomEvent }
