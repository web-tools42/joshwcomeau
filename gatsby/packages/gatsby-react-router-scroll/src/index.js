import ScrollBehaviorContext from "./ScrollBehaviorContext"
import ScrollContainer from "./ScrollContainer"
exports.ScrollContainer = ScrollContainer
exports.ScrollContext = ScrollBehaviorContext
