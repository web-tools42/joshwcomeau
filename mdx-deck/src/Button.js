import React from 'react'
import styled from 'styled-components'
import { space, color } from 'styled-system'

const Button = styled.button(
  [],
  {
    appearance: 'none',
    fontFamily: 'inherit',
    fontSize: '12px',
    fontWeight: 'bold',
    borderRadius: '4px',
    border: 'none'
  },
  space,
  color
)

Button.propTypes = {
  ...space.propTypes,
  ...color.propTypes
}

Button.defaultProps = {
  m: 0,
  px: 2,
  py: 1,
  color: 'white',
  bg: '#333'
}

export default Button
