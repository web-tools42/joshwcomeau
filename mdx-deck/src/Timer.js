import React from 'react'
import hhmmss from 'hhmmss'
import Flex from './Flex'
import Box from './Box'
import Mono from './Mono'
import Button from './Button'
class Timer extends React.Component {
  state = {
    on: false,
    time: new Date().toLocaleTimeString(),
    seconds: 0
  }

  toggle = () => {
    this.setState(state => ({ on: !state.on }))
  }

  reset = () => {
    this.setState({ seconds: 0 })
  }

  tick = () => {
    const now = new Date()
    this.setState(state => ({
      time: now.toLocaleTimeString(),
      seconds: state.on ? state.seconds + 1 : state.seconds
    }))
  }

  componentDidMount() {
    this.timer = setInterval(this.tick, 1000)
  }

  componentWillUnmount() {
    if (!this.timer) return
    clearInterval(this.timer)
  }

  render() {
    const { time, seconds, on } = this.state
    const elapsed = hhmmss(seconds)

    return (
      <Flex css={{ alignItems: 'center' }}>
        {!on &&
          seconds > 0 && (
            <Button mr={1} onClick={this.reset}>
              reset
            </Button>
          )}
        <Button bg={on ? '#600' : '#060'} onClick={this.toggle}>
          {on ? 'stop' : 'start'}
        </Button>
        <Mono px={2}>{elapsed} |</Mono>
        <Mono>{time}</Mono>
      </Flex>
    )
  }
}

export default Timer
