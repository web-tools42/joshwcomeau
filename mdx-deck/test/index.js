import React from 'react'
import { create as render } from 'react-test-renderer'
import { renderIntoDocument, Simulate } from 'react-dom/test-utils'
import 'jest-styled-components'
import Carousel from '../src/Carousel'
import Slide from '../src/Slide'
import Dots from '../src/Dots'
import Root from '../src/Root'
import GoogleFonts from '../src/GoogleFonts'
import { SlideDeck } from '../src'

const renderJSON = el => render(el).toJSON()

describe('components', () => {
  describe('Carousel', () => {
    test('renders', () => {
      const json = renderJSON(<Carousel index={1}>Hi</Carousel>)
      expect(json).toMatchSnapshot()
    })
  })

  describe('Slide', () => {
    test('renders', () => {
      const json = renderJSON(<Slide index={1}>Hi</Slide>)
      expect(json).toMatchSnapshot()
    })
  })

  describe('Dots', () => {
    test('renders', () => {
      const json = renderJSON(<Dots index={0} length={1} />)
      expect(json).toMatchSnapshot()
    })

    test('renders with index', () => {
      const json = renderJSON(<Dots index={3} length={8} />)
      expect(json).toMatchSnapshot()
    })
  })

  describe('Root', () => {
    test('renders', () => {
      const json = renderJSON(<Root>Hi</Root>)
      expect(json).toMatchSnapshot()
    })

    test('renders with font', () => {
      const json = renderJSON(
        <Root
          theme={{
            font: 'Roboto'
          }}
        />
      )
      expect(json).toHaveStyleRule('font-family', 'Roboto')
    })

    test('renders with colors', () => {
      const json = renderJSON(
        <Root
          theme={{
            colors: {
              text: 'tomato',
              background: 'black'
            }
          }}
        />
      )
      expect(json).toHaveStyleRule('color', 'tomato')
      expect(json).toHaveStyleRule('background-color', 'black')
    })
  })

  describe('GoogleFonts', () => {
    test('renders with link tag', () => {
      const json = renderJSON(
        <GoogleFonts
          theme={{
            font: '"Roboto Mono"'
          }}
        />
      )
      expect(json.type).toBe('link')
    })

    test('renders with two link tags', () => {
      const json = renderJSON(
        <GoogleFonts
          theme={{
            font: '"Roboto"',
            monospace: '"Roboto Mono"'
          }}
        />
      )
      expect(json.length).toBe(2)
    })
  })

  describe('SlideDeck', () => {
    test('renders', () => {
      const json = renderJSON(<SlideDeck slides={[]} />)
      expect(json).toMatchSnapshot(``)
    })

    test('renders with slides', () => {
      const Hello = () => <h1>Hello</h1>
      const root = render(<SlideDeck slides={[Hello]} />).root
      const hello = root.findByType(Hello)
      expect(hello)
    })

    test('initializes state from window.location.hash', () => {
      window.history.pushState(null, null, '/#2')
      const root = renderIntoDocument(<SlideDeck />)
      expect(root.state.index).toBe(2)
    })

    test('handles keydown events', () => {
      window.history.pushState(null, null, '/')
      const root = renderIntoDocument(
        <SlideDeck slides={[() => false, () => false]} />
      )
      const e = new KeyboardEvent('keydown', {
        keyCode: 39
      })
      expect(root.state.index).toBe(0)
      document.body.dispatchEvent(e)
      expect(root.state.index).toBe(1)
    })

    test('handles left arrow keydown', () => {
      window.history.pushState(null, null, '/#1')
      const root = renderIntoDocument(
        <SlideDeck slides={[() => false, () => false]} />
      )
      const e = new KeyboardEvent('keydown', {
        keyCode: 37
      })
      expect(root.state.index).toBe(1)
      document.body.dispatchEvent(e)
      expect(root.state.index).toBe(0)
    })

    test('ignoreKeyEvents does not fire handle events when set to true', () => {
      window.history.pushState(null, null, '/#1')
      const root = renderIntoDocument(
        <SlideDeck ignoreKeyEvents={true} slides={[() => false, () => false]} />
      )
      const e = new KeyboardEvent('keydown', {
        keyCode: 37
      })
      expect(root.state.index).toBe(1)
      document.body.dispatchEvent(e)
      expect(root.state.index).toBe(1)
    })

    test.skip('handles hashchange events', () => {
      window.history.pushState(null, null, '/')
      const root = renderIntoDocument(
        <SlideDeck slides={[() => false, () => false]} />
      )
      expect(root.state.index).toBe(0)
      window.location.hash = '#1'
      expect(root.state.index).toBe(1)
    })
  })
})

