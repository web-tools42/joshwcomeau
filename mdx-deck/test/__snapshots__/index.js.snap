// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Carousel renders 1`] = `
.c0 {
  overflow-x: hidden;
  width: 100%;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: .3s;
  transition-duration: .3s;
  -webkit-transform: translate3d(-100%,0,0);
  -ms-transform: translate3d(-100%,0,0);
  transform: translate3d(-100%,0,0);
}

@media print {
  .c0 {
    height: auto;
    overflow-x: visible;
  }
}

@media print {
  .c1 {
    height: auto;
    display: block;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Hi
  </div>
</div>
`;

exports[`components Dots renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 4px solid transparent;
  background-clip: padding-box;
  border-radius: 9999px;
  width: 8px;
  height: 8px;
  color: inherit;
  opacity: 0.5;
  margin: 0px;
  padding: 4px;
  color: text;
  background-color: text;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 0 1px;
}

@media print {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    color="text"
    onClick={[Function]}
    title="go to: 0"
  />
</div>
`;

exports[`components Dots renders with index 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 4px solid transparent;
  background-clip: padding-box;
  border-radius: 9999px;
  width: 8px;
  height: 8px;
  color: inherit;
  opacity: 0.5;
  margin: 0px;
  padding: 4px;
  color: text;
  background-color: text;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 0 1px;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 4px solid transparent;
  background-clip: padding-box;
  border-radius: 9999px;
  width: 8px;
  height: 8px;
  color: inherit;
  opacity: 0.125;
  margin: 0px;
  padding: 4px;
  color: text;
  background-color: text;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 0 1px;
}

@media print {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    color="text"
    onClick={[Function]}
    title="go to: 0"
  />
  <button
    className="c1"
    color="text"
    onClick={[Function]}
    title="go to: 1"
  />
  <button
    className="c1"
    color="text"
    onClick={[Function]}
    title="go to: 2"
  />
  <button
    className="c1"
    color="text"
    onClick={[Function]}
    title="go to: 3"
  />
  <button
    className="c2"
    color="text"
    onClick={[Function]}
    title="go to: 4"
  />
  <button
    className="c2"
    color="text"
    onClick={[Function]}
    title="go to: 5"
  />
  <button
    className="c2"
    color="text"
    onClick={[Function]}
    title="go to: 6"
  />
  <button
    className="c2"
    color="text"
    onClick={[Function]}
    title="go to: 7"
  />
</div>
`;

exports[`components Root renders 1`] = `
.c0 {
  color: text;
  background-color: background;
}

@media print {
  .c0 {
    font-size: 24px;
    height: auto;
  }
}

<div
  className="c0"
  color="text"
>
  Hi
</div>
`;

exports[`components Slide renders 1`] = `
.c0 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  width: 100%;
  height: 100%;
  padding-left: 32px;
  padding-right: 32px;
}

@media print {
  .c0 {
    width: 100vw;
    height: 100vh;
    page-break-after: always;
    page-break-inside: avoid;
    -webkit-print-color-adjust: exact;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    padding-left: 64px;
    padding-right: 64px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    padding-left: 128px;
    padding-right: 128px;
  }
}

<div
  className="c0"
>
  Hi
</div>
`;

exports[`components SlideDeck renders 1`] = `
Array [
  .c1 {
  overflow-x: hidden;
  width: 100%;
  height: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: .3s;
  transition-duration: .3s;
  -webkit-transform: translate3d(0%,0,0);
  -ms-transform: translate3d(0%,0,0);
  transform: translate3d(0%,0,0);
}

.c0 {
  font-family: system-ui,sans-serif;
  font-size: 16px;
  text-align: center;
  width: 100vw;
  height: 100vh;
  color: #000;
  background-color: white;
}

@media print {
  .c1 {
    height: auto;
    overflow-x: visible;
  }
}

@media print {
  .c2 {
    height: auto;
    display: block;
  }
}

@media print {
  .c0 {
    font-size: 24px;
    height: auto;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    font-size: 32px;
  }
}

<div
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="c0"
      color="text"
      height="100vh"
      mode="NORMAL"
      step={0}
      width="100vw"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
        />
      </div>
    </div>
  </div>,
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 8px;
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
}

@media print {
  .c1 {
    display: none;
  }
}

<div
    className="c0"
  >
    <div
      className="c1"
    />
  </div>,
  .c1 {
  cursor: pointer;
  width: 64px;
  height: 100vh;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
}

<div
    className="c0 c1"
    onClick={[Function]}
    role="button"
    title="Previous Slide"
  />,
  .c1 {
  cursor: pointer;
  width: 64px;
  height: 100vh;
}

.c0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
}

<div
    className="c0 c1"
    onClick={[Function]}
    role="button"
    title="Next Slide"
  />,
]
`;
